src/configfinder.coffee

```diff
@@ -28,12 +28,18 @@ findFile = (name, dir) ->
 # Possibly find CoffeeLint configuration within a package.json file.
 loadNpmConfig = (dir) ->
     fp = findFile('package.json', dir)
-    loadJSON(fp)?.coffeelintConfig  if fp
+    loadJSON(fp, 'coffeelintConfig') if fp
 
 # Parse a JSON file gracefully.
-loadJSON = (filename) ->
+loadJSON = (filename, attr = null) ->
     try
-        JSON.parse(stripComments(fs.readFileSync(filename).toString()))
+        config = JSON.parse(stripComments(fs.readFileSync(filename).toString()))
+        if attr
+            return null if not config[attr]?
+            config = config[attr]
+
+        config.__location__ = filename
```

Comment:
This is so we can remember where the config file is loaded from, so we can resolve the "extends" property relative to the location of the configfile (as opposed to the cwd)